packages/core/src/components/results/DNDAnswerContainer.tsx

```diff
@@ -0,0 +1,72 @@
+import { UniqueIdentifier } from '@dnd-kit/core'
+import React, { useContext } from 'react'
+import { ExamComponentProps } from '../..'
+import { query, queryAll } from '../../dom-utils'
+import { ResultsContext } from '../context/ResultsContext'
+import { DNDTitleAndDroppable } from '../shared/DNDTitleAndDroppable'
+import { CorrectDNDAnswers } from './CorrectDNDAnswers'
+import { ItemsState } from '../exam/DNDAnswerContainer'
+
+export const DNDAnswerContainer = ({ element, renderChildNodes }: ExamComponentProps) => {
+  const { answersByQuestionId, gradingStructure } = useContext(ResultsContext)
+  const dndAnswers = queryAll(element, 'dnd-answer').filter(e => !!query(e, 'dnd-answer-title'))
+  const dndAnswerOptions = queryAll(element, 'dnd-answer-option')
+
+  const answerOptionIdsByQuestionId: ItemsState = dndAnswers.reduce(
+    (acc, group) => {
+      const questionId = group.getAttribute('question-id')!
+      const answer = answersByQuestionId[Number(questionId)]?.value
+      return {
+        ...acc,
+        [questionId]: answer ? [Number(answer)] : [],
+        root: acc.root.filter(e => e !== Number(answer))
+      }
+    },
+    { root: dndAnswerOptions.map(e => Number(e.getAttribute('option-id')!)) }
+  )
+
+  const answerOptionsById = dndAnswerOptions.reduce(
+    (acc, el) => {
+      const optionId = el.getAttribute('option-id')!
+      return { ...acc, [optionId]: el }
+    },
+    {} as Record<UniqueIdentifier, Element>
+  )
```

Comment:
tää ois parempi näin niin ei tarvis mitään type assertionia
```suggestion
  const answerOptionsById: Record<UniqueIdentifier, Element> = dndAnswerOptions.reduce((acc, el) => {
    const optionId = el.getAttribute('option-id')!
    return { ...acc, [optionId]: el }
  }, {})
```

Reply:
👍